resteasy-rxjava2/src/main/java/org/jboss/resteasy/rxjava2/ObservableRxInvokerImpl.java

```diff
@@ -0,0 +1,267 @@
+	}
+
+	@Override
+	public <R> Observable<?> method(String name, GenericType<R> responseType)
+	{
+		return eventSourceToObservable(getEventSource(), responseType, name, null, getAccept());
+	}
+
+	@Override
+	public Observable<?> method(String name, Entity<?> entity)
+	{
+		return eventSourceToObservable(getEventSource(), String.class, name, entity, getAccept());
+	}
+
+	@Override
+	public <R> Observable<?> method(String name, Entity<?> entity, Class<R> responseType)
+	{
+		return eventSourceToObservable(getEventSource(), responseType, name, entity, getAccept());
+	}
+
+	@Override
+	public <R> Observable<?> method(String name, Entity<?> entity, GenericType<R> responseType)
+	{
+		return eventSourceToObservable(getEventSource(), responseType, name, entity, getAccept());
+	}
+
+	///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	private <T> Observable<T> eventSourceToObservable(SseEventSourceImpl sseEventSource, Class<T> clazz, String verb, Entity<?> entity, MediaType[] mediaTypes)
+	{
+		Observable<T> observable = Observable.create(
+				new ObservableOnSubscribe<T>() {
+
+					@Override
+					public void subscribe(ObservableEmitter<T> emitter) throws Exception {
+						sseEventSource.register(
+								(InboundSseEvent e) -> {T t = e.readData(clazz, ((InboundSseEventImpl) e).getMediaType()); emitter.onNext(t);},
+								(Throwable t) -> emitter.onError(t),
+								() -> emitter.onComplete());
+						synchronized (monitor)
+						{
+							if (!sseEventSource.isOpen())
+							{
+								sseEventSource.open(null, verb, entity, mediaTypes);
+							}
+						}
+					}
+				});
+		return observable;
+	}
+
+	private <T> Observable<T> eventSourceToObservable(SseEventSourceImpl sseEventSource, GenericType<T> type, String verb, Entity<?> entity, MediaType[] mediaTypes)
+	{
+		Observable<T> observable = Observable.create(
+				new ObservableOnSubscribe<T>() {
+
+					@Override
+					public void subscribe(ObservableEmitter<T> emitter) throws Exception {
+						sseEventSource.register(
+								(InboundSseEvent e) -> {T t = e.readData(type, ((InboundSseEventImpl) e).getMediaType()); emitter.onNext(t);},
+								(Throwable t) -> emitter.onError(t),
+								() -> emitter.onComplete());
+						synchronized (monitor)
+						{
+							if (!sseEventSource.isOpen())
+							{
+								sseEventSource.open(null, verb, entity, mediaTypes);
+							}
+						}
+					}
+				});
+		return observable;
+	}
+
+	private SseEventSourceImpl getEventSource()
+	{
+		SourceBuilder builder = (SourceBuilder) SseEventSource.target(syncInvoker.getTarget());
+		if (executorService != null)
+		{
+			builder.executor(executorService);
+		}
+		SseEventSourceImpl sseEventSource = (SseEventSourceImpl) builder.build();
+		sseEventSource.setAlwasyReconnect(false);
+		return sseEventSource;
+	}
+
+	private MediaType[] getAccept()
+	{
+		if (syncInvoker instanceof ClientInvocationBuilder)
```

Comment:
Is this correct? syncInvoker is always ClientInvocationBuilder ...

Reply:
@marekkopecky, you're right, but javax.ws.rs.client.SyncInvoker is an interface, so, in principle, someone could use a different implementation, right?